### DISCUSSION TITLE
docker-compose up --build on raspberry pi 4 is failing

### DISCUSSION THREAD

**Discussion by ccdanieldb** [mrlt8/docker-wyze-bridge#97] [Q&A]
Ok I installed docker and docker-compose and updated my docker-compose.yml file as per the instructions but build is failing on step 9 (update) due to the several of the below GTG errors and I do not know why.  

`GPG error: http://security.debian.org/debian-security bullseye-security InRelease: 	At least one invalid signature was encountered.`

My setup is listed below

`~/docker-wyze-bridge $ uname -a Linux 5.10.52-v7l+ #1441 SMP Tue Aug 3 18:11:56 BST 2021 armv7l GNU/Linux`

`~/docker-wyze-bridge $ docker --version Docker version 20.10.8, build 3967b7d`

I have dumped the output from `docker-compose up --build`

`	~/docker-wyze-bridge $ sudo docker-compose up --build
	
        Building wyze-bridge
	Sending build context to Docker daemon  31.74kB
	Step 1/19 : ARG ARM=1
	Step 2/19 : ARG ARCH=${ARM:+arm32v7}
	Step 3/19 : FROM ${ARCH:-amd64}/python:3.9-slim as base
	 ---> 6b23301db870
	Step 4/19 : FROM base as builder
	 ---> 6b23301db870
	Step 5/19 : ENV PYTHONUNBUFFERED=1
	 ---> Using cache
	 ---> 450732dbd869
	Step 6/19 : ARG ARM
	 ---> Using cache
	 ---> 226d6e834e78
	Step 7/19 : ARG TUTK_ARCH=${ARM:+Arm11_BCM2835_4.8.3}
	 ---> Using cache
	 ---> c9e696bd5603
	Step 8/19 : ARG RTSP_ARCH=${ARM:+armv7}
	 ---> Using cache
	 ---> 78ead2b8412d
	Step 9/19 : ARG FFMPEG_ARCH=${ARM:+armhf}
	 ---> Using cache
	 ---> e61428d77d0c
	Step 10/19 : RUN apt-get update     && apt-get install -y tar xz-utils unzip curl jq 			g++     && apt-get clean     && rm -rf /var/lib/apt/lists/*
 	---> Running in aa35c7a3d31a
	Get:1 http://deb.debian.org/debian bullseye InRelease [113 kB]
	Get:2 http://security.debian.org/debian-security bullseye-security InRelease [44.1 kB]
	Err:2 http://security.debian.org/debian-security bullseye-security InRelease
	  At least one invalid signature was encountered.
	Err:1 http://deb.debian.org/debian bullseye InRelease
	  At least one invalid signature was encountered.
	Get:3 http://deb.debian.org/debian bullseye-updates InRelease [36.8 kB]
	Err:3 http://deb.debian.org/debian bullseye-updates InRelease
	  At least one invalid signature was encountered.
	Reading package lists...
	W: GPG error: http://security.debian.org/debian-security bullseye-security InRelease: 	At least one invalid signature was encountered.
	E: The repository 'http://security.debian.org/debian-security bullseye-security InRelease' is not signed.
	W: GPG error: http://deb.debian.org/debian bullseye InRelease: At least one invalid signature was encountered.
	E: The repository 'http://deb.debian.org/debian bullseye InRelease' is not signed.
	W: GPG error: http://deb.debian.org/debian bullseye-updates InRelease: At least one invalid signature was encountered.
	E: The repository 'http://deb.debian.org/debian bullseye-updates InRelease' is not signed.
	The command '/bin/sh -c apt-get update     && apt-get install -y tar xz-utils unzip curl jq g++     && apt-get clean     && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100
	ERROR: Service 'wyze-bridge' failed to build : Build failed
	pi@ar8-pi4-8:~/docker-wyze-bridge $ `

**Comment by mrlt8** [comment]
Try `docker system prune`

**Reply by ccdanieldb** [reply]
Ok now that is cool.  After a `docker system prune`  No change.  But after a `docker system prune -a` it re-downloaded the images.  

But still same results.  I am getting several GPG errors.  

Is bullseye a new Debian release?

**Reply by mrlt8** [reply]
Yeah, we're using a Python base image that uses Debian.

I honestly don't know what's causing the error... do you have enough disk space?

You could always try using the prebuilt image with the default yaml to see if it works and build it from source if you still needed to.

**Reply by ccdanieldb** [reply]
I found this https://github.com/ArchiveTeam/warrior-dockerfile/issues/43 may be a related issue.  

This is way over my head.  But when I Use the prebuilt image: mrlt8/wyze-bridge:latest the build dose complete but  I also get some other fatal errors.  (see below)

How do I check to see if I am running at 32 or 64 bit?

   `Creating wyze-bridge ... done

     Attaching to wyze-bridge
     wyze-bridge    | Fatal Python error: init_interp_main: can't initialize time
     wyze-bridge    | Python runtime state: core initialized
     wyze-bridge    | PermissionError: [Errno 1] Operation not permitted
     wyze-bridge    | 
     wyze-bridge    | Current thread 0xb6f63010 (most recent call first):
     wyze-bridge    | <no Python frame>`

**Reply by mrlt8** [reply]
very strange. 
I just fired up my pi4 and pulled a fresh image and did a clean build, and they both ran fine.

I think your issue may be related to this: https://docs.linuxserver.io/faq

**Reply by natekspencer** [reply]
Run this on your rpi4:

`wget http://ftp.us.debian.org/debian/pool/main/libs/libseccomp/libseccomp2_2.5.1-1_armhf.deb`

`sudo dpkg -i libseccomp2_2.5.1-1_armhf.deb`

See https://github.com/linuxserver/docker-papermerge/issues/4

**Reply by ccdanieldb** [reply]
natekspencer

Running the above two commands worked.  I am not sure what it did.  Can you explain?  

Also, I had to setup my docker-compose.yml like below comment out the local image and build from source.  In addition have 8 wyzecam v2's that are already producing their own rtsp streams so I added a filter to environment "FILTER_MODEL=v3" to only allow my two v3's to stream.  

natekspencer and mrlt8,

Thank you both for all your help in getting this up and running.  

version: '2.4'
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: always
        #image: mrlt8/wyze-bridge:latest # Use a prebuilt image
        build:  # Uncomment to build from source
            context: ./app # Uncomment to build from source
            dockerfile: Dockerfile.arm # Uncomment to build for arm
        ports:
            - 1935:1935
            - 8554:8554
            - 8888:8888
        environment:
            - WYZE_EMAIL=email@ddress # Replace with wyze email
            - WYZE_PASSWORD=password # Replace with wyze password
            - FILTER_MODEL=v3

**Reply by mrlt8** [reply]
I think it was the Debian bullseye base image that was causing the issue, so the 32-bit arm image from v0.5.17+ will be using a buster image from now on.
Please let me know if you have any issues!
